net.solarnetwork.node.datum.energymeter.mock/src/net/solarnetwork/node/datum/energymeter/mock/MockEnergyMeterDatumSource.java

```diff
@@ -0,0 +1,211 @@
+package net.solarnetwork.node.datum.energymeter.mock;
+
+import java.util.Date;
+import java.util.List;
+
+import org.springframework.context.MessageSource;
+
+import net.solarnetwork.node.DatumDataSource;
+import net.solarnetwork.node.domain.GeneralNodeACEnergyDatum;
+import net.solarnetwork.node.settings.SettingSpecifier;
+import net.solarnetwork.node.settings.SettingSpecifierProvider;
+import net.solarnetwork.node.settings.support.BasicTextFieldSettingSpecifier;
+import net.solarnetwork.node.support.DatumDataSourceSupport;
+
+/**
+ * Mock plugin to be the source of values for a GeneralNodeACEnergyDatum, the
+ * datum will contain random values based around a settable center deviating by
+ * a settable deviation.
+ * 
+ * <p>
+ * This class implements {@link SettingSpecifierProvider} and
+ * {@link DatumDataSource}
+ * </p>
+ * 
+ * @author robert
+ * @version 1.0
+ */
+
+public class MockEnergyMeterDatumSource extends DatumDataSourceSupport
+		implements DatumDataSource<GeneralNodeACEnergyDatum>, SettingSpecifierProvider {
+
+	// default values
+	private String sourceId = "Mock Energy Meter";
+	private String voltagerms = "230";
+	private String frequency = "50";
+	private String resistance = "10";
+	private String inductance = "10";
+
+	private MessageSource messageSource;
+
+	private Integer apparantPow;
+	private Float phaseVolt;
+	private Float phaseCurr;
+	private Integer watt;
+	private Integer realPow;
+	private Integer reacPow;
+	private Float powfac;
+
+	@Override
+	public Class<? extends GeneralNodeACEnergyDatum> getDatumType() {
+		return GeneralNodeACEnergyDatum.class;
+	}
+
+	/*
+	 * (non-Javadoc)
+	 * 
+	 * @see net.solarnetwork.node.DatumDataSource#readCurrentDatum()
+	 * 
+	 * Returns a {@link GeneralNodeACEnergyDatum} filled with mock data. The
+	 * method uses reflection to find setters in {@link
+	 * GeneralNodeACEnergyDatum} and fill them with a number equal to the
+	 * baseReading +- a random number that goes upto deviation.
```

Comment:
A certain amount of randomness would still be a nice feature, so charts and such have a little bit of variability to them. Maybe the randomness can be tuned by a "random factor" so the randomness can be disabled, e.g. 1.0 for no randomness and > 1.0 for increasing randomness.